mlflow/models/evaluation/default_evaluator.py

```diff
@@ -540,6 +596,19 @@ def plot_pr_curve():
 
         self._log_pandas_df_artifact(per_class_metrics_collection_df, "per_class_metrics")
 
+    def _evaluate_custom_metrics(self):
+        for i, custom_metric in enumerate(self.custom_metrics):
+            # recreating the dataframe for each custom metric function call,
+            # in case the user modifies the eval_df inside their custom function.
+            eval_df = pd.DataFrame(
+                {"prediction": copy.deepcopy(self.y_pred), "target": copy.deepcopy(self.y)}
+            )
+            metric_results, _ = _evaluate_custom_metric(i, custom_metric, eval_df, self.metrics)
```

Comment:
Why is user modifying self.metrics not concerning?

Reply:
Right, that's a good point.